engine/src/widget-syntax.cpp

```diff
@@ -380,6 +380,63 @@ extern "C" MC_DLLEXPORT_DEF void MCWidgetGetClickCount(bool p_current, unsigned
 
 ////////////////////////////////////////////////////////////////////////////////
 
+extern "C" MC_DLLEXPORT_DEF void MCWidgetGetTouchIndex(MCValueRef& r_index)
+{
+    if (!MCWidgetEnsureCurrentWidget())
+        return;
+    
+    integer_t t_index;
+    if (!MCwidgeteventmanager->GetActiveTouch(t_index))
+    {
+        r_index = MCValueRetain(kMCNull);
+        return;
+    }
+    
+    MCNumberCreateWithInteger(t_index, (MCNumberRef&)r_index);
+}
+
+extern "C" MC_DLLEXPORT_DEF void MCWidgetGetTouchPosition(MCValueRef& r_point)
+{
+    if (!MCWidgetEnsureCurrentWidget())
+        return;
+    
+    integer_t t_index;
+    MCPoint t_position;
+    if (!MCwidgeteventmanager->GetActiveTouch(t_index) ||
+        !MCwidgeteventmanager->GetTouchPosition(t_index, t_position))
+    {
+        r_point = MCValueRetain(kMCNull);
+        return;
+    }
+
+    /* UNCHECKED */ MCCanvasPointCreateWithMCGPoint(MCWidgetMapPointFromGlobal(MCcurrentwidget, MCPointToMCGPoint(t_position)), (MCCanvasPointRef&)r_point);
+}
+
+extern "C" MC_DLLEXPORT_DEF void MCWidgetGetNumberOfTouches(uinteger_t& r_count)
+{
+    if (!MCWidgetEnsureCurrentWidget())
+        return;
+    
+    r_count = MCwidgeteventmanager->GetTouchCount();
+}
+
+extern "C" MC_DLLEXPORT_DEF void MCWidgetGetPositionOfTouch(integer_t p_id, MCValueRef& r_point)
```

Comment:
@runrevmark to reduce confusion (presuming we keep the `index` concept) we should rename this `p_index`